hazelcast/src/test/java/com/hazelcast/jet/core/JobUploadTest.java

```diff
@@ -0,0 +1,271 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.impl.HazelcastBootstrap;
+import com.hazelcast.internal.util.Sha256Util;
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.JetService;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.config.JetConfig;
+import com.hazelcast.jet.impl.JetClientInstanceImpl;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static com.hazelcast.internal.util.Sha256Util.calculateSha256Hex;
+import static java.util.Collections.emptyList;
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mockStatic;
+
+@Category({QuickTest.class})
+public class JobUploadTest extends JetTestSupport {
+
+    @Test
+    public void sha256() throws IOException, NoSuchAlgorithmException {
+        Path jarPath = getJarPath();
+        String sha256Hex = calculateSha256Hex(jarPath);
+        assertEquals("bba07be19c71bfe5fd51dc681f807ab212efd4d6e7c3f6380dbfcbdcb5deb073", sha256Hex);
+
+    }
+
+    @Test
+    public void test_client_jarUpload_whenResourceUploadIsNotEnabled() {
+        HazelcastInstance hazelcastInstance = createHazelcastInstance();
+        HazelcastInstance client = createHazelcastClient();
+        JetService jetService = client.getJet();
+        List<String> jobParameters = emptyList();
+        assertThrows(JetException.class, () ->
+                jetService.uploadJob(getJarPath(),
+                        null,
+                        null,
+                        null,
+                        jobParameters)
+        );
+
+        assertEqualsEventually(() -> jetService.getJobs().size(), 0);
+        hazelcastInstance.shutdown();
+    }
+
+    @Test
+    public void test_member_jarUpload_whenResourceUploadIsNotEnabled() {
```

Comment:
Using `TestInClusterSupport` as a base class you can get rid of the duplicated tests for the same thing on member/client.

Reply:
I can't do it because the tests are mocking client specific sha256.And there is only one test for the member side, so I think it is better if it stays this way. But thanks for the info. I had not seen TestInClusterSupport class